voila/handler.py

```diff
@@ -140,9 +140,16 @@ def _jinja_cell_generator(self, nb, kernel_id):
         nb, resources = ClearOutputPreprocessor().preprocess(nb, {'metadata': {'path': self.cwd}})
         ep = VoilaExecutePreprocessor(config=self.traitlet_config)
 
+        stop_execution = False
         with ep.setup_preprocessor(nb, resources, km=km):
             for cell_idx, cell in enumerate(nb.cells):
-                res = ep.preprocess_cell(cell, resources, cell_idx, store_history=False)
+                if stop_execution:
+                    break
+                try:
+                    res = ep.preprocess_cell(cell, resources, cell_idx, store_history=False)
+                except TimeoutError:
+                    res = (cell, resources)
+                    stop_execution = True
```

Comment:
What about
```suggestion
            for cell_idx, cell in enumerate(nb.cells):
                try:
                    res = ep.preprocess_cell(cell, resources, cell_idx, store_history=False)
                except TimeoutError:
                    res = (cell, resources)
                    stop_execution = True
                    return
                finally:
                    yield res[0]
```

Reply:
Would that mean we don't `yield res[0]` for the cell where we stop the execution?

Reply:
finally *always* run, otherwise, the test would have picked that up right?